bundles/org.dataflowanalysis.analysis.converter/src/org/dataflowanalysis/analysis/converter/DataFlowDiagramConverter.java

```
@@ -0,0 +1,432 @@
+                        }
+                    }
+                }
+            }
+            return builder.toString().trim();
+        }
+        return null;
+
+    }
+
+    // Currently only supports True or False
+    private boolean getTermValue(Term term) throws IllegalArgumentException {
+        if (term instanceof TRUE)
+            return true;
+        if (term instanceof NOT) {
+            return !getTermValue(((NOT) term).getNegatedTerm());
+        }
+        throw new IllegalArgumentException();
+    }
+
+    private void parseBehavior(Node node, Pin outpin, String lines, DataFlowDiagram dfd, DataDictionary dd) {
+        String[] behaviorStrings = lines.split("\n");
+        var behavior = node.getBehaviour();
+        for (String behaviorString : behaviorStrings) {
+            if (behaviorString.contains("forward ")) {
+                String packet = behaviorString.split(" ")[1];
+                Pin inpin = null;
+                for (Flow flow : dfd.getFlows()) {
+                    if (flow.getDestinationNode() == node) {
+                        if (flow.getEntityName().equals(packet)) {
+                            inpin = flow.getDestinationPin();
+                        }
+                    }
+                }
+
+                var assignment = ddFactory.createForwardingAssignment();
+                assignment.setOutputPin(outpin);
+                assignment.getInputPins().add(inpin);
+                behavior.getAssignment().add(assignment);
+            } else if (behaviorString.contains("set ")) {
+                String regex = "\\bset\\s+(\\S+)\\s+=\\s+(.+)\\b";
+
+                Pattern pattern = Pattern.compile(regex);
+                Matcher matcher = pattern.matcher(behaviorString);
+                matcher.find();
+
+                String variable = matcher.group(1);
```

Indexing into groups is unsafe, if the amount of groups is not checked before.
Please return an error, if the number of groups is incorrect